server/admin/test_groups.go

```diff
@@ -270,18 +265,15 @@ func (r *TestGroupCtx) WriteTests(db db.DBContext, tests []*tests.LazyTest, over
 		if err != nil {
 			return errors.Wrapf(err, "test %v output", test.Name)
 		}
-		vars = append(vars, test.Name, test.TestGroupID, input, output)
-	}
-	res, err := db.Exec(fmt.Sprintf("INSERT INTO tests(name, test_group_id, input, output) VALUES %s", strings.Join(terms, ", ")), vars...)
-	if err != nil {
-		return errors.WithStack(err)
-	}
-	id, err := res.LastInsertId()
-	if err != nil {
-		return errors.WithStack(err)
-	}
-	for i, test := range tests {
-		test.ID = int(id) - len(tests) + i + 1
+		if _, err := db.Exec(
+			"INSERT INTO tests(name, test_group_id, input, output) VALUES (?, ?, ?, ?)",
+			test.Name,
+			test.TestGroupID,
+			input,
+			output,
+		); err != nil {
+			return errors.Wrapf(err, "inserting test `%s`", test.Name)
+		}
```

Comment:
Note for future me: We upload tests one-by-one so that we don't have to store all the tests' content in memory at once. This helps with big test-sets that adds up to gigabytes of text.